Add cluster-reset docs (#417)
rancher/rke2: docs/backup_restore.md

In this section, you'll learn how to create backups of the rke2 cluster data and to restore the cluster from backup.

> 

Reviewer: Add a note here about the path prefix `/var/lib/rancher/rke2` being configurable by the `--data-dir` flag. If folks change the flag they will need to correct the path.

The way it reads now, folks are going to go looking for a /server/ path in the root of the filesystem.

### Creating Snapshots

Snapshots are enabled by default.

> The snapshot directory defaults to `/server/db/snapshots`.

Reviewer: ```suggestion
The snapshot directory defaults to `/var/lib/rancher/rke2/server/db/snapshots`.
```

To configure the snapshot interval or the number of retained snapshots, refer to the [options.](#options)

## Cluster Reset

> RKE2 enables a feature to reset the cluster to one member cluster by passing `--cluster-reset` flag, when passing this flag to rke2 server it will reset the cluster with the same data dir in place, the data directory for etcd exists in `/server/db/etcd`, this flag can be passed in the events of quorum loss in the cluster.

Reviewer: ```suggestion
RKE2 enables a feature to reset the cluster to one member cluster by passing `--cluster-reset` flag, when passing this flag to rke2 server it will reset the cluster with the same data dir in place, the data directory for etcd exists in `/var/lib/rancher/rke2/server/db/etcd`, this flag can be passed in the events of quorum loss in the cluster.
```

To pass the reset flag, first you need to stop RKE2 service if its enabled via systemd:

```
systemctl stop rke2-server
rke2 server --cluster-reset
```

> **Result:**  A message in the logs says that RKE2 can be restarted without the flags. Start rke2 again and it should start rke2 as a 1 member cluster.

Reviewer 2: ```suggestion
**Result:**  A message in the logs say that RKE2 can be restarted without the flags. Start rke2 again and it should start rke2 as a 1 member cluster.
```

 
### Restoring a Cluster from a Snapshot

> When RKE2 is restored from backup, the old data directory will be moved to `/server/db/etcd-old-%date%/`. Then RKE2 will attempt to restore the snapshot by creating a new data directory, then starting etcd with a new RKE2 cluster with one etcd member.

Reviewer: ```suggestion
When RKE2 is restored from backup, the old data directory will be moved to `/var/lib/rancher/rke2/server/db/etcd-old-%date%/`. Then RKE2 will attempt to restore the snapshot by creating a new data directory, then starting etcd with a new RKE2 cluster with one etcd member.
```

To restore the cluster from backup, run RKE2 with the `--cluster-reset` option, with the `--cluster-reset-restore-path` also given:

```
rke2 server \
  --cluster-reset \
  --cluster-reset-restore-path=<PATH-TO-SNAPSHOT>
```

**Result:**  A message in the logs says that RKE2 can be restarted without the flags. Start RKE2 again and should run successfully and be restored from the specified snapshot.

> Note that when rke2 starts with cluster-reset it adds a file called reset-flag to /server/db/ directory, so its not possible to run two cluster-reset in a row without deleting that file manually.

Reviewer 2: ```suggestion
Note that when rke2 starts with cluster-reset it adds a file called reset-flag to /server/db/ directory, so it's not possible to run two cluster-reset in a row without deleting that file manually.
```

> Note that when rke2 starts with cluster-reset it adds a file called reset-flag to /server/db/ directory, so its not possible to run two cluster-reset in a row without deleting that file manually.

Reviewer: ```suggestion
When RKE2 resets the cluster, it creates a file at `/var/lib/rancher/rke2/server/db/etc/reset-file`. If you want to reset the cluster again, you will need to delete this file.
```

### Options

> These options can be passed in with the command line, or in the configuration file.  which may be easier to use.

Reviewer 2: ```suggestion
These options can be passed in with the command line, or in the configuration file.  Which may be easier to use.
```

> These options can be passed in with the command line, or in the configuration file.  which may be easier to use.

Reviewer: ```suggestion
These options can be passed in with the command line, or in the configuration file (which may be easier to use).
```
